src/ui/components/Overlay/index.js

```diff
@@ -1,80 +1,101 @@
+  children: React.Element<any>,
+  id: string,
+  onEscapeOverlay?: () => void,
+  visibleOnLoad?: boolean,
+|};
+
+type UIStateType = {|
+  visible: boolean,
+|};
+
+const initialUIState: UIStateType = { visible: false };
+
+type InternalProps = {|
+  ...Props,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|};
 
+export class OverlayBase extends React.Component<InternalProps> {
   static defaultProps = {
     visibleOnLoad: false,
   };
 
-  constructor(props) {
-    super(props);
-    this.state = { visible: props.visibleOnLoad };
+  componentDidMount() {
+    const { visibleOnLoad } = this.props;
+    this.props.setUIState({ visible: visibleOnLoad });
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.visibleOnLoad !== undefined) {
-      this.setState({ visible: nextProps.visibleOnLoad });
+  componentWillReceiveProps(nextProps: InternalProps) {
+    const { uiState } = this.props;
+    const { visibleOnLoad: visibleOnLoadNew } = nextProps;
+
+    if (
+      visibleOnLoadNew !== undefined &&
+      visibleOnLoadNew !== uiState.visible
+    ) {
+      this.props.setUIState({ visible: visibleOnLoadNew });
     }
   }
 
-  onClickBackground = () => {
+  onClickBackground = (event: SyntheticEvent<any>) => {
+    event.preventDefault();
     if (this.props.onEscapeOverlay) {
       this.props.onEscapeOverlay();
     }
+
     this.hide();
   };
 
-  hide() {
-    this.setState({ visible: false });
-  }
-
-  show() {
-    this.setState({ visible: true });
-  }
-
-  toggle() {
-    this.setState({ visible: !this.state.visible });
-  }
+  hide = () => {
+    this.props.setUIState({ visible: false });
+  };
 
   render() {
-    const { children, className } = this.props;
+    const { children, className, id, uiState } = this.props;
+
+    invariant(children, 'The children property is required');
+    invariant(id, 'The id property is required');
 
     return (
       <div
         className={makeClassName('Overlay', className, {
-          'Overlay--visible': this.state.visible,
+          'Overlay--visible': uiState.visible,
         })}
-        ref={(ref) => {
-          this.overlayContainer = ref;
-        }}
       >
         <div
-          onClick={this.onClickBackground}
-          ref={(ref) => {
-            this.overlayBackground = ref;
-          }}
           className="Overlay-background"
+          onClick={this.onClickBackground}
           role="presentation"
         />
-        <div
-          className="Overlay-contents"
-          ref={(ref) => {
-            this.overlayContents = ref;
-          }}
-        >
-          {children}
-        </div>
+        <div className="Overlay-contents">{children}</div>
       </div>
     );
   }
 }
+
+export const extractId = (ownProps: Props) => {
+  return ownProps.id;
+};
+
+const Overlay: React.ComponentType<Props> = compose(
+  withUIState({
+    fileName: __filename,
+    extractId,
+    initialState: initialUIState,
+  }),
```

Comment:
I know it said before it was working without resetting this before but it seems to be okay with the default now 🤷‍♂️